frontend/src/pages/ResourceSelectorV2.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2023 The Kubeflow Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Separate the resource selector between v1 and v2 to avoid breaking current v1 behavior
+// TODO(jlyaoyuli): consider to merge 2 selectors together (change updatedSelection() in v1)
```

Comment:
+1. 
What you have is fine for now. But I do think avoiding code duplication is an important factor to maintain a healthy code base. As discussed offline, the alternative could be having converters to/from `BaseResource` like the api_converter in the BE code. 
We can target such cleanups post-GA.

Reply:
Cleanups for this one is in #9693